### ISSUE TITLE
Add ability to map string fields to numeric fields

### ISSUE THREAD

**Issue by bluebob** [influxdata/telegraf#2791]
### Proposal:
Introduce a numeric "cluster_health" or "cluster_status" value

### Current behavior:
Current implementation has a status field which is a text object, having values "green", "yellow" or "red".

### Desired behavior:
Add a numeric representation of this field, called cluster_health, cluster_status or just health. I suggest a simple mapping of 0, 1 and 2 for green,yellow and red.

### Use case: [Why is this important (helps with prioritizing requests)]
The "SignalFX" plugin for CollectD already has this behaviour. Having a numeric field permits Grafana presentation of state with time, so showing _when_ a cluster went into a degraded or failed state. This can be very useful in tracking down root cause. Current text-based representation has to be converted to a number in order to do this, which is cumbersome.
Alerting rules are also simplified, since != 0 means intervention is needed

**Comment by phemmer** [comment]
Just for references:
Here's a grafana feature request which is to allow colorizing table / singlestat based on string values: https://github.com/grafana/grafana/issues/3601
Here's a influxdb feature request which would allow conditional substitution of values ("case when" stuff): https://github.com/influxdata/influxdb/issues/5383
Here's a telegraf feature request that would allow consitional substitution of values: https://github.com/influxdata/telegraf/issues/2667 (there's other issues with similar discussions about the end goal there, but can't find them at the moment)

And just for clarification, I'm not saying this request should be closed, just throwing out some related stuff.

**Comment by bluebob** [comment]
Thanks. That info will be really handy as a "stop gap". 👍

**Comment by danielnelson** [comment]
I think it is best to implement this with some enhancements to the #2667 proposal.  I'll write a follow up over there.

**Comment by danielnelson** [comment]
Changing my mind on this, I guess it doesn't make sense to have a general purpose transform plugin so we would want a "enum" or "case" processor for this.

Here is my first idea for how to configure the processor.  I don't love it, so hopefully we can improve on the idea.
```toml
[[processors.enum]]
  namepass = ["elasticsearch"]
  [[processors.enum.fields]]
    key = "health"
    [processors.enum.table]
      green = 0
      yellow = 1
      red = 2
```

**Comment by phemmer** [comment]
Kind of a crazy ass idea, but what about embedding kapacitor in telegraf? We wouldn't put the whole thing in, just the stream processing engine.
There are a lot of open issues about people (myself included) wanting to tweak the input that comes from plugins. It seems like instead of introducing a ton of very specific capabilities to handle each individual need, we could put in a general purpose processor that could meet any need. And then instead of writing one from scratch, use one that already exists.

**Comment by danielnelson** [comment]
I'd love to do it but I guess it's a fair amount of work.

**Comment by dupondje** [comment]
This is also fixed! :)

**Comment by danielnelson** [comment]
Closed in #3772
